### ISSUE TITLE
Specifying a CLOUDTRAIL_AUDIT_LOGS parameter of an Event Data Store in another account fails to load the session logs

### ISSUE THREAD

**Issue by reidca** [aws-samples/iam-identity-center-team#202]
In an organization it is common to have an existing cloudtrail lake setup with an existing cloudtrail event data store. 
In order to reduce the costs of setting up another one I opted to specify the arn of the existing CloudTrail Event Data Store in the parameter CLOUDTRAIL_AUDIT_LOGS .

However, when looking at the session logs for an active elevate session, the logs do not load.

Checking the lambda cloud watch log for the lambda function `teamgetLogs-main` I see the following:

```
2024-03-13T15:41:40.308Z	634f9a5f-ed6d-4e34-9eee-79877b050b83	INFO	Error EventDataStoreNotFoundException: The event data store was not found.
    at de_EventDataStoreNotFoundExceptionRes (/var/task/node_modules/@aws-sdk/client-cloudtrail/dist-cjs/protocols/Aws_json1_1.js:3238:23)
    at de_StartQueryCommandError (/var/task/node_modules/@aws-sdk/client-cloudtrail/dist-cjs/protocols/Aws_json1_1.js:2577:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /var/task/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
    at async /var/task/node_modules/@aws-sdk/middleware-signing/dist-cjs/awsAuthMiddleware.js:14:20
    at async /var/task/node_modules/@aws-sdk/middleware-retry/dist-cjs/retryMiddleware.js:27:46
    at async /var/task/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26
    at async start_query (file:///var/task/index.js:147:22)
    at async Runtime.handler (file:///var/task/index.js:159:19) {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    requestId: '94a36766-e9b6-42b1-b086-0fd71298f1a2',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  __type: 'EventDataStoreNotFoundException'
}
```

Checking the code for the same lambda I see how the Cloudtrail client is being created:
`const client = new CloudTrailClient({ region: REGION });`

So the client is being created in the same account as TEAM is deployed to and of course cannot access the CloudTrail Event Data Store in another account. 

I have had a brief look at the documentation for the CloudTrail Event Data store and the [IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_aws-services-that-work-with-iam.html#swwiam_footnotes_cloudtrail:~:text=CloudFront%20Developer%20Guide.-,AWS%20CloudTrail,and%20channels.%20CloudTrail%20doesn%27t%20support%20tag%2Dbased%20access%20controls%20for%20trails.,-Amazon%20CloudWatch) I **do not** see that is supports resource policies for Event Data Stores (only channels).

Therefore I believe the only fix for this would be to assume a role inside the account where the CloudTrail Event Data Store exists if it is in a different account other than where TEAM is deployed. This would need some changes and probably some new parameters but I am sure it is doable.

For now, I would suggest that the TEAM documentation shows that it is only possible to connect to an existing CloudTrail Event Data Store if it resides in the same account and region (since the region is read from the same environment variable that is used for all other deployments).

**Comment by reidca** [comment]
I have spent a little more time looking at this and the original report was incorrect. When CloudTrail is setup using as delegated admin account, the CloudTrail Event Data store is available in the delegated admin account which is the account that TEAM deploys so. Therefore the issue is not an account problem. However, if the CloudTrail Event Data Store is setup in the a different region other than the one that TEAM is deployed to (which in our case is true) then the code fails to find the event data store.

There are a couple of small changes needed to this:

1. Lambda function `teamqueryLogs-main` uses the environment variable REGION to establish the CloudTrail client. This variable is passed right through from the initial parameters.sh file. It is not possible to change the value to the correct region (where CloudTrail EDS can be found) since this parameter is used for many other parts of the TEAM app. As a short term, runtime fix I simply changed the value of the environment variable in the Lambda console to the correct one where the EDS can be found. Since this env variable is not used for anything else in the code this works.
2.  Lambda function `teamgetLogs-main` uses the environment variable REGION to establish the CloudTrail client as well as to sign the requests sent to the graph QL endpoint. Therefore I had to add a new environment variable CLOUDTRAIL_REGION and set this manually in the lambda console to the region where the CloudTrail EDS can be found. I then modified the code that creates the CloudTrail client to use this: `const client = new CloudTrailClient({ region: CLOUDTRAIL_REGION });`. This allows the cloudtrail client to use a different region than the one where the GraphQL endpoint is. 

With these changes I can now see my session logs in the TEAM application. 

I think the long term fix for this is to add specific parameters for the regions for CloudTrail, SES and the other TEAM components. This will allow complex setups with multi-region deployments to succeeed.
